On process 1.6.27.0, sys/wait.h is included unconditionally which breaks builds on wasm32-wasi

Commit 5c98f59 moved the `#if defined(HAVE_FORK)` guard in cbits/posix/runProcess.c but left
`#include <sys/wait.h>` outside the guard. 

On wasm32-wasi, sys/wait.h does not exist and HAVE_FORK is not defined, causing a build failure.

This was found on the x86_64-linux-alpine3_23-wasm-cross_wasm32-wasi-release GHC CI job.

I can verify with the following output I suddenly got:

```
[3 of 8] Compiling System.Process.Posix ( System/Process/Posix.hs, dist/build/System/Process/Posix.o, dist/build/System/Process/Posix.dyn_o )
System/Process/Posix.hs:230:23: warning: [GHC-63394] [-Wdeprecations]
    In the use of ‘installHandler’
    (imported from System.Posix.Signals):
    "operation will throw 'IOError' "unsupported operation" (CPP guard: @#if HAVE_SIGNAL_H@)"
    |
230 |           old_int  <- installHandler sigINT  Ignore Nothing
    |                       ^^^^^^^^^^^^^^

System/Process/Posix.hs:231:23: warning: [GHC-63394] [-Wdeprecations]
    In the use of ‘installHandler’
    (imported from System.Posix.Signals):
    "operation will throw 'IOError' "unsupported operation" (CPP guard: @#if HAVE_SIGNAL_H@)"
    |
231 |           old_quit <- installHandler sigQUIT Ignore Nothing
    |                       ^^^^^^^^^^^^^^

System/Process/Posix.hs:245:16: warning: [GHC-63394] [-Wdeprecations]
    In the use of ‘installHandler’
    (imported from System.Posix.Signals):
    "operation will throw 'IOError' "unsupported operation" (CPP guard: @#if HAVE_SIGNAL_H@)"
    |
245 |           _ <- installHandler sigINT  old_int  Nothing
    |                ^^^^^^^^^^^^^^

System/Process/Posix.hs:246:16: warning: [GHC-63394] [-Wdeprecations]
    In the use of ‘installHandler’
    (imported from System.Posix.Signals):
    "operation will throw 'IOError' "unsupported operation" (CPP guard: @#if HAVE_SIGNAL_H@)"
    |
246 |           _ <- installHandler sigQUIT old_quit Nothing
    |                ^^^^^^^^^^^^^^

System/Process/Posix.hs:323:5: warning: [GHC-63394] [-Wdeprecations]
    In the use of ‘createPipe’
    (imported from System.Posix.IO, but defined in System.Posix.IO.Common):
    "operation will throw 'IOError' "unsupported operation" (CPP guard: @#if HAVE_PIPE@)"
    |
323 |     (readfd, writefd) <- Posix.createPipe
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

System/Process/Posix.hs:330:4: warning: [GHC-63394] [-Wdeprecations]
    In the use of ‘createPipe’
    (imported from System.Posix.IO, but defined in System.Posix.IO.Common):
    "operation will throw 'IOError' "unsupported operation" (CPP guard: @#if HAVE_PIPE@)"
    |
330 |    (Fd readfd, Fd writefd) <- Posix.createPipe
    |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

System/Process/Posix.hs:342:25: warning: [GHC-63394] [-Wdeprecations]
    In the use of ‘getProcessGroupIDOf’
    (imported from System.Posix.Process, but defined in System.Posix.Process.Common):
    "operation will throw 'IOError' "unsupported operation" (CPP guard: @#if HAVE_GETPID@)"
    |
342 |                 pgid <- getProcessGroupIDOf h
    |                         ^^^^^^^^^^^^^^^^^^^

System/Process/Posix.hs:343:17: warning: [GHC-63394] [-Wdeprecations]
    In the use of ‘signalProcessGroup’
    (imported from System.Posix.Signals):
    "operation will throw 'IOError' "unsupported operation" (CPP guard: @#if HAVE_SIGNAL_H@)"
    |
343 |                 signalProcessGroup sigINT pgid
    |                 ^^^^^^^^^^^^^^^^^^

[4 of 8] Compiling System.Process.Internals ( System/Process/Internals.hs, dist/build/System/Process/Internals.o, dist/build/System/Process/Internals.dyn_o )
[5 of 8] Compiling System.Process.CommunicationHandle.Internal ( dist/build/System/Process/CommunicationHandle/Internal.hs, dist/build/System/Process/CommunicationHandle/Internal.o, dist/build/System/Process/CommunicationHandle/Internal.dyn_o )
[6 of 8] Compiling System.Process   ( System/Process.hs, dist/build/System/Process.o, dist/build/System/Process.dyn_o )
System/Process.hs:142:1: warning: [GHC-66111] [-Wunused-imports]
    The import of ‘System.IO.Error’ is redundant
      except perhaps to import instances from ‘System.IO.Error’
    To import instances alone, use: import System.IO.Error()
    |
142 | import System.IO.Error (mkIOError, ioeSetErrorString)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

System/Process.hs:743:5: warning: [GHC-63394] [-Wdeprecations]
    In the use of ‘getProcessID’
    (imported from System.Posix.Process, but defined in System.Posix.Process.Common):
    "operation will throw 'IOError' "unsupported operation" (CPP guard: @#if HAVE_GETPID@)"
    |
743 |     getProcessID
    |     ^^^^^^^^^^^^

[7 of 8] Compiling System.Process.CommunicationHandle ( System/Process/CommunicationHandle.hs, dist/build/System/Process/CommunicationHandle.o, dist/build/System/Process/CommunicationHandle.dyn_o )
[8 of 8] Compiling System.Cmd       ( System/Cmd.hs, dist/build/System/Cmd.o, dist/build/System/Cmd.dyn_o )
cbits/posix/runProcess.c:12:10: error:
     fatal error: 'sys/wait.h' file not found
       12 | #include <sys/wait.h>
          |          ^~~~~~~~~~~~
   |
12 | #include <sys/wait.h>
   |          ^

1 error generated.
`wasm32-wasi-clang' failed in phase `C Compiler'. (Exit code: 1)
```

Will attempt to downgrade until this works?

--- a/cbits/posix/runProcess.c
+++ b/cbits/posix/runProcess.c
@@ -5,6 +5,11 @@
    ------------------------------------------------------------------------- */
 
 #include "runProcess.h"
+
+// Callers are stubbed out on wasm32: runInteractiveProcess (System/Process/Posix.hs),
+// terminateProcess, getProcessExitCode, waitForProcess (System/Process.hs).
+#if !defined(wasm32_HOST_ARCH)
+
 #include "common.h"
 
 #include <unistd.h>
@@ -271,3 +276,5 @@ waitForProcess (ProcHandle handle, int *pret)
 
     return -1;
 }
+
+#endif // !wasm32_HOST_ARCH

--- a/process.cabal
+++ b/process.cabal
@@ -94,7 +94,7 @@ library
 
     ghc-options: -Wall
 
-    build-depends: base      >= 4.12.0.0 && < 4.23,
+    build-depends: base      >= 4.12.0.0 && < 4.24,
                    directory >= 1.1 && < 1.4,
                    deepseq   >= 1.1 && < 1.6
 

--- a/test/process-tests.cabal
+++ b/test/process-tests.cabal
@@ -22,7 +22,7 @@ common process-dep
 
 custom-setup
   setup-depends:
-    base      >= 4.10 && < 4.23,
+    base      >= 4.10 && < 4.24,
     directory >= 1.1  && < 1.4,
     filepath  >= 1.2  && < 1.6,
     Cabal     >= 2.4  && < 3.18,

